events/api.py

```diff
@@ -1762,6 +1762,15 @@ def _filter_event_queryset(queryset, params, srs=None):
     """
     # Filter by string (case insensitive). This searches from all fields
     # which are marked translatable in translation.py
+    val = params.get('rolling', None)
+    if val and validate_bool(val, 'rolling'):
+        passed = Event.objects.exclude(super_event=None).filter(start_time__lte=datetime.now()
+                                                                                .astimezone(pytz.timezone('UTC'))  # noqa E127
+                                                                                ).values_list('super_event__pk')  # noqa E124
+        not_yet = Event.objects.exclude(super_event=None).filter(start_time__gte=datetime.now()
+                                                                                 .astimezone(pytz.timezone('UTC'))  # noqa E127    
+                                                                                 ).values_list('super_event__pk')  # noqa E124
+        queryset = queryset.filter(Q(pk__in=Subquery(passed)) & Q(pk__in=Subquery(not_yet)))
```

Comment:
From the Jira ticket https://helsinkisolutionoffice.atlassian.net/browse/LINK-273:
> Esimerkki 1: etsin lapselleni kevääksi sopivaa harrastusta tapahtumat.hel.fi-sivustolta 16.2.2021 klo 10.21. Haluan suodattaa näkyviin vain ne harrastukset, joiden ensimmäinen kerta on tulevaisuudessa eli 16.2.2021 klo alkaen. En siis halua näkyviin harrastuksia/kursseja, jotka ovat alkaneet jo tammikuussa.

If I understand that correctly, it means the super events that does not have a single sub event in the past. So the whole super event with all its sub events would all be in the future. Events like that could be wanted to be searched so there wouldn't be any already missed (sub) events.

Reply:
I think that this does resolve the 2nd need, (but not the 1st one):
> Käyttötapaus 2: Nykyisessä toteutuksessa (https://latest-events.test.kuva.hel.ninja/fi/courses) harrastusta etsivänä henkilönä näen yhdessä näkymässä kaikki meneillään olevat ja tulevat harrastukset. Myös mahdollisuus suodatukseen tällä tavalla halutaan säilyttää.

> Esimerkki 2: etsin harrastuksia tapahtumat.hel.fi-sivustolta 16.2.2021. Haluan nähdä kaiken harrastustarjonnan, sekä ennen 16.2.2021 alkaneet että tulevaisuudessa alkavat.